src/httpserver.cpp

```diff
@@ -288,11 +282,9 @@ static void http_request_cb(struct evhttp_request* req, void* arg)
 
     // Dispatch to worker thread
     if (i != iend) {
-        std::auto_ptr<HTTPWorkItem> item(new HTTPWorkItem(hreq.release(), path, i->handler));
+        std::unique_ptr<HTTPWorkItem> item(new HTTPWorkItem(hreq.release(), path, i->handler));
         assert(workQueue);
-        if (workQueue->Enqueue(item.get()))
-            item.release(); /* if true, queue took ownership */
-        else
+        if (!workQueue->Enqueue(std::move(item)))
             item->req->WriteReply(HTTP_INTERNAL, "Work queue depth exceeded");
```

Comment:
The actual moving happens in `unique_ptr`'s ctor. The function just needs to accept a `unique_ptr&&`, then it can freely move that into a new `unique_ptr` or not.